nixos/modules/hardware/video/nvidia.nix

```diff
@@ -180,6 +189,25 @@ in
     environment.systemPackages = [ nvidia_x11.bin nvidia_x11.settings ]
       ++ lib.filter (p: p != null) [ nvidia_x11.persistenced ];
 
+    systemd.services = let
+      nvidiaService = type: {
+        description = "NVIDIA system ${type} actions";
+        before = [ "systemd-${type}.service" ];
+
+        path = with pkgs; [ coreutils nvidia_x11.bin ];
+        serviceConfig = {
+          Type = "oneshot";
+        };
+
+        script = ''
+          logger -t ${type} -s "nvidia-${type}.service"
```

Comment:
systemd should automatically log when it's enabling services. You should be able to just put the `nvidia-sleep.sh` call into `serviceConfig.ExecStart`.